mm2src/coins/lp_coins.rs

```diff
@@ -985,6 +990,35 @@ pub struct RefundPaymentArgs<'a> {
     pub watcher_reward: bool,
 }
 
+#[derive(Debug)]
+pub struct RefundTakerPaymentArgs<'a> {
+    pub payment_tx: &'a [u8],
+    pub time_lock: u64,
+    pub maker_pub: &'a [u8],
+    pub tx_type_with_secret_hash: SwapTxTypeWithSecretHash<'a>,
+    pub swap_unique_data: &'a [u8],
+    pub watcher_reward: bool,
+    pub dex_fee: &'a DexFee,
+    /// Additional reward for maker (premium)
+    pub premium_amount: BigDecimal,
+    /// Actual volume of taker's payment
+    pub trading_amount: BigDecimal,
+}
+
+impl<'a> From<RefundTakerPaymentArgs<'a>> for RefundPaymentArgs<'a> {
+    fn from(args: RefundTakerPaymentArgs<'a>) -> RefundPaymentArgs<'a> {
+        RefundPaymentArgs {
+            payment_tx: args.payment_tx,
+            time_lock: args.time_lock,
+            other_pubkey: args.maker_pub,
+            tx_type_with_secret_hash: args.tx_type_with_secret_hash,
+            swap_contract_address: &None,
+            swap_unique_data: args.swap_unique_data,
+            watcher_reward: args.watcher_reward,
+        }
+    }
+}
```

Comment:
> Maybe we should have RefundTakerPaymentArgsV2 and RefundMakerPaymentArgsV2 and leave the original RefundPaymentArgs to be used only in legacy swaps. What do you think?

totally agree, its better to leave legacy args and have separate taker / maker types for all v2 operations. will update structures

Reply:
hm, so Ive provided new maker args to MakerSwapOpsV2 methods: `RefundMakerPaymentTimelockArgs`, `RefundMakerPaymentSecretArgs` 
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/adb69d59ff56a7e8205b21edd1e8305ae6434bfb

But we still need to impl From Taker and Maker Refund args  for `RefundPaymentArgs`, as in UTXO we re-use `utxo_common::refund_htlc_payment` inside these methods: `refund_maker_payment_v2_timelock`, `refund_taker_funding_timelock`, `refund_combined_taker_payment`. And at the same time we use `refund_htlc_payment` function in utxo legacy code.

Reply:
> But we still need to impl From Taker and Maker Refund args for RefundPaymentArgs, as in UTXO we re-use utxo_common::refund_htlc_payment inside these methods

Instead of from implementation, just do the conversions inside the functions, e.g.
```rust
    async fn refund_maker_payment_v2_timelock(
        &self,
        args: RefundMakerPaymentTimelockArgs<'_>,
    ) -> Result<Self::Tx, TransactionErr> {
        let args = RefundPaymentArgs {
            payment_tx: args.payment_tx,
            time_lock: args.time_lock,
            other_pubkey: args.taker_pub,
            tx_type_with_secret_hash: args.tx_type_with_secret_hash,
            swap_contract_address: &None,
            swap_unique_data: args.swap_unique_data,
            watcher_reward: args.watcher_reward,
        };
        utxo_common::refund_htlc_payment(self.clone(), args).await
    }
```

Reply:
> Instead of from implementation, just do the conversions inside the functions

That is what I had :D https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169#discussion_r1710273715

I think your suggestion about `From` for `RefundPaymentArgs` also makes sense, so I will come back to manual conversion inside function.

Reply:
Done here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/722b8071fdd818d26496492a195a7b87eb1225dc